sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/UnnecessaryImportsCheck.kt

```diff
@@ -52,22 +53,31 @@ class UnnecessaryImportsCheck : AbstractCheck() {
 
         val (references, arrayAccesses, kDocLinks) = collectReferences(file)
         val bindingContext = context.bindingContext
+
+        val unresolvedImports = bindingContext.diagnostics.noSuppression()
+            .filter { it.psiFile == file }
+            .mapNotNull { it.psiElement.getParentOfType<KtImportDirective>(false) }
+
         val groupedReferences = references.groupBy { reference ->
             reference.importableSimpleName()
         }
 
         val arrayAccessesImportsFilter by lazy { getArrayAccessImportsFilter(arrayAccesses, bindingContext) }
 
-        analyzeImports(file, groupedReferences, kDocLinks, arrayAccessesImportsFilter, context)
+        analyzeImports(file, groupedReferences, kDocLinks, unresolvedImports, arrayAccessesImportsFilter, context)
     }
 
     private fun analyzeImports(
         file: KtFile,
         groupedReferences: Map<String?, List<KtReferenceExpression>>,
         kDocLinks: Collection<String>,
+        unresolvedImports: List<KtImportDirective>,
         arrayAccessesImportsFilter: (KtImportDirective) -> Boolean,
         context: KotlinFileContext
-    ) = file.importDirectives.filter { imp ->
+    ) = file.importDirectives
+        // 0. Filter out unresolved imports, to avoid FPs in case of incomplete semantic
+        .filter { it !in unresolvedImports }
+        .filter { imp ->
         // 1. Filter out & report all imports that import from kotlin.* or the same package as our file
         if (imp.isImportedImplicitlyAlready(file.packageDirective?.name)) {
```

Comment:
The indentation is off here, due to a mix of using 
```
... }
.foo { bar }
```
and
```
...
}.foo {
    bar
}
```
styles. I think we should stay consistent, then the indentation can also stay consistent.